engine/sync_manager.go

```diff
@@ -397,72 +356,72 @@ func (m *syncManager) Update(exchangeName string, p currency.Pair, a asset.Item,
 		return fmt.Errorf("%v %w", syncType, errUnknownSyncItem)
 	}
 
-	m.mux.Lock()
-	defer m.mux.Unlock()
+	k := currencyPairKey{
+		AssetType: a,
+		Exchange:  exchangeName,
+		Pair:      p,
+	}
 
-	for x := range m.currencyPairs {
-		if m.currencyPairs[x].Exchange == exchangeName &&
-			m.currencyPairs[x].Pair.Equal(p) &&
-			m.currencyPairs[x].AssetType == a {
-			switch syncType {
-			case SyncItemTicker:
-				origHadData := m.currencyPairs[x].Ticker.HaveData
-				m.currencyPairs[x].Ticker.LastUpdated = time.Now()
-				if err != nil {
-					m.currencyPairs[x].Ticker.NumErrors++
-				}
-				m.currencyPairs[x].Ticker.HaveData = true
-				m.currencyPairs[x].Ticker.IsProcessing = false
-				if atomic.LoadInt32(&m.initSyncCompleted) != 1 && !origHadData {
-					removedCounter++
-					log.Debugf(log.SyncMgr, "%s ticker sync complete %v [%d/%d].",
-						exchangeName,
-						m.FormatCurrency(p).String(),
-						removedCounter,
-						createdCounter)
-					m.initSyncWG.Done()
-				}
-				return nil
-			case SyncItemOrderbook:
-				origHadData := m.currencyPairs[x].Orderbook.HaveData
-				m.currencyPairs[x].Orderbook.LastUpdated = time.Now()
-				if err != nil {
-					m.currencyPairs[x].Orderbook.NumErrors++
-				}
-				m.currencyPairs[x].Orderbook.HaveData = true
-				m.currencyPairs[x].Orderbook.IsProcessing = false
-				if atomic.LoadInt32(&m.initSyncCompleted) != 1 && !origHadData {
-					removedCounter++
-					log.Debugf(log.SyncMgr, "%s orderbook sync complete %v [%d/%d].",
-						exchangeName,
-						m.FormatCurrency(p).String(),
-						removedCounter,
-						createdCounter)
-					m.initSyncWG.Done()
-				}
-				return nil
-			case SyncItemTrade:
-				origHadData := m.currencyPairs[x].Trade.HaveData
-				m.currencyPairs[x].Trade.LastUpdated = time.Now()
-				if err != nil {
-					m.currencyPairs[x].Trade.NumErrors++
-				}
-				m.currencyPairs[x].Trade.HaveData = true
-				m.currencyPairs[x].Trade.IsProcessing = false
-				if atomic.LoadInt32(&m.initSyncCompleted) != 1 && !origHadData {
-					removedCounter++
-					log.Debugf(log.SyncMgr, "%s trade sync complete %v [%d/%d].",
-						exchangeName,
-						m.FormatCurrency(p).String(),
-						removedCounter,
-						createdCounter)
-					m.initSyncWG.Done()
-				}
-				return nil
-			}
+	c, exists := m.currencyPairs[k]
+	if !exists {
+		return fmt.Errorf("%w for %s %s %s", errCouldNotSyncNewData, k.Exchange, k.Pair, k.AssetType)
+	}
+
+	c.locks[syncType].Lock()
+	defer c.locks[syncType].Unlock()
+
+	if c.trackers[syncType] == nil {
+		c.trackers[syncType] = &syncBase{}
+	}
+	s := c.trackers[syncType]
+
+	if !s.IsUsingWebsocket {
+		s.IsUsingWebsocket = true
+		s.IsUsingREST = false
+		n := "Ticker"
+		if syncType == SyncItemOrderbook {
+			n = "Orderbook"
```

Comment:
For REST only `.\gocryptotrader -tickersync=false -orderbooksync=true` Seems like syncing just orderbooks it alerts that ticker sync is complete. `Okx ticker sync complete LTC-USD-230623 [1/6].`

Reply:
Okay. Probably not this line, since this would be giving:
` Websocket re-enabled, switching from rest to websocket`
and uppercase Ticker.
Again, will reproduce and fix or ask for help. 🙏